io/js/src/main/scala/fs2/io/file/FilesPlatform.scala

```diff
@@ -321,7 +321,10 @@ private[fs2] trait FilesCompanionPlatform {
       readStream(path, chunkSize, Flags.Read)(_.setStart(start.toDouble).setEnd((end - 1).toDouble))
 
     def realPath(path: Path): F[Path] =
-      F.fromPromise(F.delay(fsPromisesMod.realpath(path.toString))).map(Path(_))
+      F.fromPromise(F.delay(fsPromisesMod.realpath(path.toString))).map(Path(_)).recoverWith {
+        case NoSuchFileException(e) =>
+          F.raiseError(e)
+      }
```

Comment:
I _think_ this is equivalent to:
```scala
.adaptError { case NoSuchFileException(ex) => ex }
```

Reply:
Aha! Nice! I've completely forgotten about this one :)

Reply:
Adapted it :)